main.py

```diff
@@ -362,7 +362,7 @@ def warm_models():
             do_whisper("client/3sec.flac", "medium", beam_size, "transcribe", False, "en")
         if models.whisper_model_large is not None:
             do_whisper("client/3sec.flac", "large", beam_size, "transcribe", False, "en")
-        if sv_model is not None:
+        if models.sv_model is not None:
```

Comment:
Speaking of which! The voice authentication functionality is very much a work in progress. For example, there is an extremely bizarre issue with the voice authentication processor and model being exposed through the model object that I'd really like to work through (it's a real stumper - it just doesn't fundamentally doesn't make any sense). This is what this issue is related to and it's a good fix for the time being but I would prefer to make all models accessible through the model object.

I should be committing a script today that enables users to make their own voice embeddings as long as the provided audio files meet certain requirements (minimum of 10 seconds duration, likely checks for audio level, potentially with normalization and some other processing).

This is another case where testing with my individual voice likely doesn't translate to broader users, voices, and recordings. If you're interested in this functionality we'd love to see at least one additional data point of a user that has a positive experience with voice authentication. I'm especially excited about this functionality and I think (with some additional broader support) it has the potential to really enable all kinds of very interesting and powerful things:

- It already acts as a type of "firewall" against unknown speakers. If your voice doesn't match a known speaker with your configured probability WIS will refuse to do ASR and Willow essentially shuts down for the given wake session. This makes it impossible for an unrecognized speaker to provide any commands to your configured command endpoint.

- It fundamentally enables the ability for users to define rules for which speakers can execute which commands. For example, preventing children from controlling things they shouldn't (I have a voice command to control a fireplace - YIKES).

- It also fundamentally enables the ability to prevent multiple wake of devices in proximity. There isn't a scenario I can imagine where you would want the same speaker to wake and provide commands to multiple Willow devices simultaneously. Being able to essentially "lock" command processing to one device for a given speaker and session is very powerful.

Reply:
Agree this looks very cool and useful! Exactly the thoughts I had (I have a 3 year old who is already trying to issue voice commands to stuff to control lights and media 🤣 ) Would be happy to play around with this when it is ready, I should have my S3 Box tomorrow so I can test the full thing end-to-end 😃